### DISCUSSION TITLE
CronExpression.GetTimeBefore and CronExpression.GetFinalFireTime

### DISCUSSION THREAD

**Discussion by StrangeW** [quartznet/quartznet#1278] [General]
Hello

I want to know when these 2 methods will be available

PS: Is it possible to add to the comment of GetTimeBefore "NOT YET IMPLEMENTED:"

**Comment by lahma** [comment]
There's no timeline for such changes and/or features. You can always help out by submitting PRs to fix things though 👍🏻

**Reply by StrangeW** [reply]
Thank you for this information.

I'll see what I can do

**Reply by lahma** [reply]
There's also a PR on Java side which might answer to your needs: https://github.com/quartz-scheduler/quartz/pull/421 . If it gets merged, the logic should be synchronized.
